### PR TITLE
Fix reads for GZIP compressed Hive Text.

### LINKED ISSUE #8419: [BUG] Hive Text reader fails for GZIP compressed input
When GZIP compressed Hive Text tables are read via the Spark RAPIDS plugin, the following failure occurs on large reads:
```
Caused by: java.io.IOException: Cannot seek in DeflateCodec compressed stream
        at org.apache.hadoop.mapreduce.lib.input.LineRecordReader.initialize(LineRecordReader.java:113)
        at org.apache.spark.sql.execution.datasources.HadoopFileLinesReader.<init>(HadoopFileLinesReader.scala:65)
        at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.<init>(GpuTextBasedPartitionReader.scala:218)
        at com.nvidia.spark.rapids.CSVPartitionReaderBase.<init>(GpuCSVScan.scala:302)
        at org.apache.spark.sql.hive.rapids.GpuHiveDelimitedTextPartitionReader.<init>(GpuHiveTableScanExec.scala:496)
        at org.apache.spark.sql.hive.rapids.GpuHiveTextPartitionReaderFactory.buildColumnarReader(GpuHiveTableScanExec.scala:470)
        at com.nvidia.spark.rapids.PartitionReaderIterator$.$anonfun$buildReader$1(PartitionReaderIterator.scala:59)
        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:127) 
        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:187)
        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:104)
        at org.apache.spark.sql.hive.rapids.GpuHiveTableScanExec$$anon$1.hasNext(GpuHiveTableScanExec.scala:372)
...
```
This does not occur for smaller reads. It happens because GPU Hive Text reader deems its input splittable by default.

Ideally, the compression codec for each input file needs to be evaluated for "splittability". It is conceivable that the input contains a mix of differently compressed files.

NV bug id: 4129487

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #8419.

This commit fixes how the GPU Hive Text reader determines whether the input file can be split, for reads.
Before this change, the input was deemed splittable by default. This leads to GZIP compressed Hive Text input to be split between tasks, causing read failures like the following:
```
Caused by: java.io.IOException: Cannot seek in DeflateCodec compressed stream
        at org.apache.hadoop.mapreduce.lib.input.LineRecordReader.initialize(LineRecordReader.java:113)
        at org.apache.spark.sql.execution.datasources.HadoopFileLinesReader.<init>(HadoopFileLinesReader.scala:65)
        at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.<init>(GpuTextBasedPartitionReader.scala:218)
        at com.nvidia.spark.rapids.CSVPartitionReaderBase.<init>(GpuCSVScan.scala:302)
        at org.apache.spark.sql.hive.rapids.GpuHiveDelimitedTextPartitionReader.<init>(GpuHiveTableScanExec.scala:496)
```
This commit adds logic to examine the compression codec of each input file, and determine whether it is splittable.

### CODE REVIEW

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
Build

File: integration_tests/src/main/python/hive_delimited_text_test.py [hunk 1]
```diff
@@ -422,6 +422,31 @@ def create_hive_table_with_custom_timestamp_format(spark, property_location):
...
+
+    def create_table_with_compressed_files(spark):
+        spark.range(100000)\
+            .selectExpr("id",
+                        "cast(id as string) id_string")\
+            .repartition(1).write.format("hive").saveAsTable(table_name)
+
+    # Create Hive Text table with compression enabled.
+    with_cpu_session(create_table_with_compressed_files,
+                     {"hive.exec.compress.output": "true"})
```
Comment (tgravescs) [inline integration_tests/src/main/python/hive_delimited_text_test.py:441]: gzip is guaranteed here?  or should we set compression codec explicitly?  If it supports multiple, do we have tests for them?

**Review by revans2 — COMMENTED** [review]
Looks good, but I have the same comment as Tom

File: integration_tests/src/main/python/hive_delimited_text_test.py (hunk 1 above)
Comment (mythrocks) [reply integration_tests/src/main/python/hive_delimited_text_test.py:441]: That's a good question. I've only seen GZIP/deflate. 

Let me check if I can swap the codec.

Comment (mythrocks) [reply integration_tests/src/main/python/hive_delimited_text_test.py]: I've added tests for DEFLATE, Gzip and Bzip2. There are other conceivable codecs, but I think this sample covers the reasonable gamut.

**Comment by mythrocks** [comment]
Build

**Review by revans2 — APPROVED** [review]

**Pull request merged**